src/cpp/src/speculative_decoding/continuous_batching_for_speculative_decoding_impl.cpp

```diff
@@ -337,5 +337,16 @@ void ContinuousBatchingPipeline::ContinuousBatchingForSpeculativeDecodingImpl::m
             to_generate |= request->can_generate_tokens();
         }
     }
+
+    for (auto& request : m_requests) {
```

Comment:
I suppose we initially ignore EOS tokens for draft models, why are they removed here? It should not affect results of main model, should they?

Reply:
It was decided to not add part after EOS for draft model according to this ticket https://jira.devtools.intel.com/browse/CVS-164477 . It affects results of main model. What I saw is that if we have a stop_token, the generation result can contain it and some tokens after it, with these changes it will be nothing after stop token